test/application.test.ts

```diff
@@ -278,4 +281,94 @@ describe('Application', () => {
       expect(spy).toHaveBeenCalled()
     })
   })
+
+  describe('auth cache', () => {
+    let scopeInstall: nock.Scope
+    let scopeData: nock.Scope
+
+    beforeEach(() => {
+      delete process.env.INSTALLATION_TOKEN_TTL
+
+      const cache = cacheManager.caching({
+        store: 'memory',
+        ttl: 60 * 60 // 1 hour
+      })
+      app = new Application({ cache } as any)
+      app.app = () => 'app-bearer-authorization-token'
+
+      nock.cleanAll()
+      scopeInstall = nock('https://api.github.com')
+        .post('/installations/1/access_tokens')
+        .reply(200, { token: 'installation-bearer-authorization-token' })
+      scopeData = nock('https://api.github.com')
+        .matchHeader('authorization', 'token installation-bearer-authorization-token')
+        .get('/orgs/myorg')
+        .reply(200, {})
+    })
+
+    afterEach(() => {
+      expect(scopeInstall.isDone()).toEqual(true)
+      expect(scopeData.isDone()).toEqual(true)
+    })
+
+    it('requests an installation token once for one event', async () => {
+      // Receive first event
+      app.on('test.foo', async context => {
+        await context.github.orgs.get({ org: 'myorg' })
+      })
+
+      await app.receive(event)
+    })
+
+    it('requests an installation token once for two events', async () => {
+      // Receive first event
+      app.on('test.foo', async context => {
+        await context.github.orgs.get({ org: 'myorg' })
+      })
+      await app.receive(event)
+
+      // Receive second event
+      const scopeInstallTwo = nock('https://api.github.com')
+        .post('/installations/1/access_tokens')
+          .reply(200, { token: 'token-should-not-be-requested' })
+      const scopeDataTwo = nock('https://api.github.com')
+        .matchHeader('authorization', 'token installation-bearer-authorization-token')
+        .get('/orgs/myorg')
+          .reply(200, {})
+      await app.receive(event)
+
+      // we should have not requested a second token, and just used the first one
+      expect(scopeInstallTwo.isDone()).toEqual(false)
+      expect(scopeDataTwo.isDone()).toEqual(true)
+    })
+
+    it('requests an installation token once for each event if not cached', async () => {
+      // Only cache token for 1 second
+      process.env.INSTALLATION_TOKEN_TTL = '1'
+
+      // Receive first event
+      app.on('test.foo', async context => {
+        await context.github.orgs.get({ org: 'myorg' })
+      })
+      await app.receive(event)
+
+      // Sleep for 2 seconds to let token cache expire
+      const sleep = async () => new Promise(resolve => setTimeout(resolve, 1200))
```

Comment:
Your comment mentions sleeping for 2 seconds but your code sleeps for [approximately] 1.2 seconds instead. It seems like the two values should be brought into alignment, likely just by updating the comment's accuracy, or by just removing the specific duration mention from the comment altogether, e.g. `// Sleep for long enough to let token cache expire`.

Reply:
Have amended, see below commit

Reply:
Thanks! :+1: